client/README.md

```diff
@@ -192,3 +192,1454 @@ For example:
+2. Making the search bar functional
+
+# Trip Detail
+
+## Description
+
+This page provides users with details about an upcoming trip.
+
+## URL Route
+
+```
+/TripDetail
+```
+
+###### Current Status
+
+This page only loads mock data for a specific trip: Hawaii 2020.
+
+This page should render live data depending on which trip the user clicked on from the "Book A Trip" page. No links are live or lead to any other page within the app.
+
+This page is responsive. It renders on mobile devices without breaking any components. The main content renders justified to the left of the screen, though, and would look better centered.
+
+###### Future Development
+
+Future development should focus on:
+
+1. Centering components when they render on smaller screens.
+2. Rendering details from live trip data.
+3. The "Book Trip" button should redirect users to a registration page for that specific trip.
+   <Br>
+
+## API FETCH CALLS/ROUTES:
+
+## /groups
+
+Returns a list of all groups in the database.
+
+##### Request Example
+
+```JS
+HTTP Methods: POST, GET
+fetch(`${BASE_URL}/groups`, {
+        headers: {
+          Authorization: `Token ${localStorage.getItem("token")}`
+        }
+ 	})
+```
+
+If post request, expects the following request body format
+
+```JSON
+{"title": "yourtitle",
+"description": "a description"}
+```
+
+###### Current Status
+
+The view is functional, allowing for viewing of current groups and creation of new ones, if request is authenticated.
+
+###### Future Development
+
+This route is complete. It is ready to be accessed by the front end.
+
+## /groups/<int:pk>
+
+Returns a specific instance of the model that matches the passed pk
+
+##### Request Example
+
+```JS
+HTTP Methods: GET, UPDATE, DELETE
+fetch(`${BASE_URL}/groups/<int:pk>`,{
+  headers:
+    {Authorization: `Token ${localStorage.getItem("token")}` } })
+```
+
+if UPDATE request body should include an object with the name of the field, you want to update set to its new value ex.
+
+```JSON
+{"title": "King of England"}
+```
+
+###### Current Status
+
+Route is functional, allowing for GET UPDATE and DELETE on individual instances
+
+###### Future Development
+
+See Future Development in this models list route.
+
+One thing to consider: users shouldn't be able to delete or update groups, just their association to them.
+
+## /media
+
+Returns a list of all media in the database.
+
+##### Request Example
+
+```JS
+HTTP Methods: POST, GET
+fetch(`${BASE_URL}/media`, {
+  headers: {
+    Authorization: `Token ${localStorage.getItem("token")}` } })
+```
+
+If a post request, expects the following request body format
+
+```JSON
+{
+"time": Date,
+"image": Image,
+"title": "yourtitle",
+"description": "a description"
+}
+```
+
+Optionally, the request can include
+
+```JSON
+{
+  "author": Foreign Key of accounts.Profile,
+"hashtag": Foreign Key of community.Hashtag,
+"group": Foreign Key of community.Group
+}
```

Comment:
You need to either place `Foreign Key of accounts.Profile,` in quotes or provide a sample value that goes there and then have a legend below that explains what to pass in.

Example:
```JSON
{
  "author": "<Foreign Key of accounts.Profile>",
"hashtag": "<Foreign Key of community.Hashtag>",
"group": "<Foreign Key of community.Group>"
}
```

or 
Example:
```JSON
{
  "author": "1",
"hashtag": "3",
"group": "15"
}
```
|  Field Name | Value  |
|---|---|
| author  |   Foreign Key of accounts.Profile |
|  hashtag |  Foreign Key of community.Hashtag |
|  group |  Foreign Key of community.Group |